lib/wal/wal.c

```diff
@@ -113,23 +113,25 @@ wal_throttle_sensor_set(struct wal *wal, uint64_t bufsz, uint64_t buflen)
  * This avoids overloading the WAL IO layer with numerous flushes when IOs are
  * backed up due to a slow IO backend.
  */
-HSE_ALWAYS_INLINE static void
+static HSE_ALWAYS_INLINE void
 wal_flush_wait(struct wal *wal, struct wal_flush_stats *stats, uint8_t pct)
 {
     INVARIANT(wal && stats);
 
     const int flushc = stats->bufcnt;
+    assert(flushc > 0);
```

Comment:
Please don't intersperse code and decls.  Looks like if we didn't use a VLA and you use an unsigned bufcnt/flushc then you could just assert(flushc < MAX) after the decls, right?

Reply:
This is fixed in my upcoming push as I removed the VLA..